jetty-annotations/src/main/java/org/eclipse/jetty/annotations/ContainerInitializerAnnotationHandler.java

```diff
@@ -28,13 +32,20 @@
  */
 public class ContainerInitializerAnnotationHandler extends AbstractHandler
 {
-    final ContainerInitializer _initializer;
-    final Class _annotation;
+    final ServletContainerInitializerHolder _holder;
+    final Class<?> _annotation;
 
-    public ContainerInitializerAnnotationHandler(ContainerInitializer initializer, Class annotation)
+    @Deprecated
+    public ContainerInitializerAnnotationHandler(ContainerInitializer initializer, Class<?> annotation)
+    {
+        _holder = null;
+        _annotation = null;
```

Comment:
If we're not going to support this constructor maybe we should throw `UnsupportedOperationException` or something?

Currently this will allow this to be created with a null _holder and _annotation and it will just throw NPE whenever it is used.

Or maybe we could do something like this instead to make it kind of work:
```java
@Deprecated
public ContainerInitializerAnnotationHandler(ContainerInitializer initializer, Class<?> annotation)
{
    this(new ServletContainerInitializerHolder(initializer.getTarget()), annotation);
}
```

Reply:
FWIW I reinstated the use of the old ContainerInitializer, so the ContainerInitializerAnnotationHandler now supports both.